### PR TITLE
build_defs/pkg/make_deb.py needs python2

### LINKED ISSUE #79806: make bazel-release fails with python3
`make bazel-release` fails with the following error

```
dims@rex:~/go/src/k8s.io/kubernetes$ make bazel-release
INFO: Analyzed target //build/release-tars:release-tars (2960 packages loaded, 41335 targets configured).
INFO: Found 1 target...
ERROR: /home/dims/go/src/k8s.io/kubernetes/build/debs/BUILD:197:2: MakeDeb build/debs/kube-proxy__amd64.deb failed (Exit 1) make_deb failed: error executing command
  (cd /home/dims/.cache/bazel/_bazel_dims/8b714d02207463747ec9588937c2cb2f/sandbox/linux-sandbox/10127/execroot/io_k8s_kubernetes && \
  exec env - \
  bazel-out/host/bin/external/bazel_tools/tools/build_defs/pkg/make_deb '--output=bazel-out/k8-fastbuild/bin/build/debs/kube-proxy__amd64.deb' '--changes=bazel-out/k8-fastbuild/bin/build/debs/kube-proxy__amd64.changes' '--data=bazel-out/k8-fastbuild/bin/build/debs/kube-proxy-data-amd64.tar' '--package=kube-proxy' '--architecture=amd64' '--maintainer=Kubernetes Authors <kubernetes-dev+release@googlegroups.com>' '--version=@bazel-out/k8-fastbuild/bin/build/version' '--description=Kubernetes Service Proxy' '--depends=iptables (>= 1.4.21)' '--depends=iproute2' '--homepage=http://kubernetes.io' '--distribution=unstable' '--urgency=medium')
Execution platform: @bazel_tools//platforms:host_platform

Use --sandbox_debug to see verbose messages from the sandbox: make_deb failed: error executing command
  (cd /home/dims/.cache/bazel/_bazel_dims/8b714d02207463747ec9588937c2cb2f/sandbox/linux-sandbox/10127/execroot/io_k8s_kubernetes && \
  exec env - \
  bazel-out/host/bin/external/bazel_tools/tools/build_defs/pkg/make_deb '--output=bazel-out/k8-fastbuild/bin/build/debs/kube-proxy__amd64.deb' '--changes=bazel-out/k8-fastbuild/bin/build/debs/kube-proxy__amd64.changes' '--data=bazel-out/k8-fastbuild/bin/build/debs/kube-proxy-data-amd64.tar' '--package=kube-proxy' '--architecture=amd64' '--maintainer=Kubernetes Authors <kubernetes-dev+release@googlegroups.com>' '--version=@bazel-out/k8-fastbuild/bin/build/version' '--description=Kubernetes Service Proxy' '--depends=iptables (>= 1.4.21)' '--depends=iproute2' '--homepage=http://kubernetes.io' '--distribution=unstable' '--urgency=medium')
Execution platform: @bazel_tools//platforms:host_platform

Use --sandbox_debug to see verbose messages from the sandbox
Traceback (most recent call last):
  File "/home/dims/.cache/bazel/_bazel_dims/8b714d02207463747ec9588937c2cb2f/sandbox/linux-sandbox/10127/execroot/io_k8s_kubernetes/bazel-out/host/bin/external/bazel_tools/tools/build_defs/pkg/make_deb.runfiles/bazel_tools/tools/build_defs/pkg/make_deb.py", line 367, in <module>
    main(FLAGS(sys.argv))
  File "/home/dims/.cache/bazel/_bazel_dims/8b714d02207463747ec9588937c2cb2f/sandbox/linux-sandbox/10127/execroot/io_k8s_kubernetes/bazel-out/host/bin/external/bazel_tools/tools/build_defs/pkg/make_deb.runfiles/bazel_tools/tools/build_defs/pkg/make_deb.py", line 339, in main
    version=GetFlagValue(FLAGS.version),
  File "/home/dims/.cache/bazel/_bazel_dims/8b714d02207463747ec9588937c2cb2f/sandbox/linux-sandbox/10127/execroot/io_k8s_kubernetes/bazel-out/host/bin/external/bazel_tools/tools/build_defs/pkg/make_deb.runfiles/bazel_tools/tools/build_defs/pkg/make_deb.py", line 311, in GetFlagValue
    flagvalue = flagvalue.decode('utf-8')
AttributeError: 'str' object has no attribute 'decode'
----------------
Note: The failure of target @bazel_tools//tools/build_defs/pkg:make_deb (with exit code 1) may have been caused by the fact that it is a Python 2 program that was built in the host configuration, which uses Python 3. You can change the host configuration (for the entire build) to instead use Python 2 by setting --host_force_python=PY2.

If this error started occurring in Bazel 0.27 and later, it may be because the Python toolchain now enforces that targets analyzed as PY2 and PY3 run under a Python 2 and Python 3 interpreter, respectively. See https://github.com/bazelbuild/bazel/issues/7899 for more information.
----------------
Target //build/release-tars:release-tars failed to build
INFO: Elapsed time: 109.791s, Critical Path: 64.64s
INFO: 4994 processes: 4994 linux-sandbox.
FAILED: Build did NOT complete successfully
Makefile:638: recipe for target 'bazel-release' failed
make: *** [bazel-release] Error 1
```

### PR DESCRIPTION
**Pull request by dims**
Force us to use python2 especially for scenarios where both python2 and
3 are both installed as `make bazel-release` fails currently.

Change-Id: I70bc45909ca87e5cf7ccdbddf60740a8a3e7e58b

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #79806

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by BenTheElder — APPROVED** [review]
xref: https://github.com/bazelbuild/bazel/issues/7359
We should probably try to get to python 3 this year, but this is a documented solution to an incompatible bazel defaulting change in the meantime.
/lgtm
/approve

**Comment by dims** [comment]
@BenTheElder the `/approve` did not kick in for some reason ... thanks!

File: build/root/.bazelrc [hunk 1]
```diff
@@ -1,5 +1,8 @@
 startup --expand_configs_in_place
 
+# Force us to use python2
+build --host_force_python=PY2
```
Comment (fejta) [inline build/root/.bazelrc]: Which things need py2? Can we link to issues like we did for https://github.com/kubernetes/test-infra/blob/master/.bazelrc#L8-L10

Comment (BenTheElder) [reply build/root/.bazelrc]: off the top of my head https://github.com/bazelbuild/rules_docker/issues/842#issuecomment-508263187 (hey look, you're on this thread! :upside_down_face: ), not sure if that's everything though.

Comment (BenTheElder) [reply build/root/.bazelrc]: https://github.com/bazelbuild/rules_docker#setup would also be a good note ("NEW: Starting with Bazel 0.27.0, you also need to add to your .bazelrc file the following: ...")

**Comment by dims** [comment]
@fejta looks like @BenTheElder already answered on my behalf :) can you please peek again?

**Comment by BenTheElder** [comment]
@dims I think erick is looking to put a comment in the file like https://github.com/kubernetes/test-infra/pull/13240/files so we know to remove / deal with it later

**Comment by dims** [comment]
@BenTheElder @fejta Done!

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve
Thanks @dims!

**Comment by BenTheElder** [comment]
> @BenTheElder the /approve did not kick in for some reason ... thanks!

oh er, I forgot I don't actually have `/approve` for `build/` 🤷‍♂

**Comment by spiffxp** [comment]
/lgtm
/assign @cblecker 
for the /approve (I'm only a reviewer for build/)

File: build/root/.bazelrc (hunk 1 above)
Comment (fejta) [reply build/root/.bazelrc]: ref https://github.com/bazelbuild/bazel/issues/7899 https://github.com/bazelbuild/rules_docker/issues/842
/lgtm
/approve

**Comment by fejta** [comment]
> we know to remove / deal with it later

Right at some point I expect rules_docker to fix this. So if we see that issue closed perhaps we can try removing this flag and seeing if things work again, and if not note any additional issues that require it.

**Pull request merged**